utils/types.go

```diff
@@ -193,9 +195,13 @@ func parseHeaders(headers []string) (http.Header, error) {
 func GetKonnectClient(httpClient *http.Client, config KonnectConfig) (*konnect.Client,
 	error) {
 	address := CleanAddress(config.Address)
+	headers, err := parseHeaders(config.Headers)
+	if err != nil {
+		return nil, fmt.Errorf("parsing headers: %w", err)
+	}
 	client, err := konnect.NewClient(httpClient, konnect.ClientOpts{
 		BaseURL: address,
-	})
+	}, headers)
```

Comment:
Let's make changes only in this file and bring changes that are in `konnect/client.go` into this file. 
This way we can keep the "client with header" logic within this file and re-use it across the Kong and Konnect clients.